src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/outertypefilename/Example5.java

```diff
@@ -0,0 +1,18 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="OuterTypeFilename"/>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.outertypefilename;
+
+// xdoc section -- start
+// public class Example1 {
+// // This class is commented out
+// }
+
+class Foo5 { // violation, 'The name of the outer type and the file do not match.'
+
+}
```

Comment:
Documentation uses `record` for this example. Is there any reason why you have decided to go with class instead?
You probably were facing issues because you weren't placing it in `resources-noncompilable`.

Refer to the PR #13275  for reference. (Particularly look for how `Example1.java` is used from `resources-noncompilable`, update the test accordingly.)